presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -399,34 +405,59 @@ public boolean trySetMemoryReservation(long newMemoryReservation)
 
     public synchronized boolean isMemoryRevokingRequested()
     {
-        return memoryRevokingRequestedFuture.isDone();
+        return memoryRevokingRequested;
     }
 
     /**
      * Returns how much revocable memory will be revoked by the operator
      */
-    public synchronized long requestMemoryRevoking()
-    {
-        boolean alreadyRequested = isMemoryRevokingRequested();
-        if (!alreadyRequested && revocableMemoryReservation > 0) {
-            memoryRevokingRequestedFuture.set(null);
-            return revocableMemoryReservation;
+    public long requestMemoryRevoking()
+    {
+        long revocableMemory = 0L;
```

Comment:
"revokedMemory" would be a better name, since Operator may have >0 revocable memory that is already being revoked and this variable would still be zero.